### PR TITLE
Improve AndroidIDE #3

### LINKED ISSUE #214: Crash After close preview xml layout
**Describe the bug**
when closing preview for xml layout got crash

**To Reproduce**
open project -> open layout.xml, click three dots on right side , then preview xml layout , then press back on phone

**Expected behavior**
should be going back to xml layout editor

**Screenshots**

**Details (please complete the following information):**
 - Device: Redmi Note 5 pro
 - OS: Android 10
 - AndroidIDE Version : 202

**Additional context**

```
AndroidIDE crash report
Manufacturer: Xiaomi
Device: Redmi Note 5 Pro
App version: 2.0.2-beta (202)

 Stacktrace: 
android.util.SuperNotCalledException: Activity {com.itsaky.androidide/com.itsaky.androidide.DesignerActivity} did not call through to super.onDestroy()
	at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4930)
	at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:4971)
	at android.app.servertransaction.DestroyActivityItem.execute(DestroyActivityItem.java:44)
	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2016)
	at android.os.Handler.dispatchMessage(Handler.java:107)
	at android.os.Looper.loop(Looper.java:214)
	at android.app.ActivityThread.main(ActivityThread.java:7356)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:924)
```

### PR DESCRIPTION
**Pull request by MrIkso**
Added log filter feature
Fixed #214

### CODE REVIEW

File: app/src/main/java/com/itsaky/androidide/utils/SortUtils.java [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        break;
+      case SORT_SIZE:
+        Arrays.sort(items, Comparator_SIZE);
+        content.clear();
+
+        for (String a : items) {
+          if (new File(current + "/" + a).isDirectory()) content.add(index++, a);
+          else content.add(a);
+        }
+        break;
```
Comment (itsaky) [inline app/src/main/java/com/itsaky/androidide/utils/SortUtils.java:48]: You can create a simple `Comparator` which does the sorting based on size and type of `File`. See [this](https://github.com/itsaky/AndroidIDE/blob/222fa49ee15d561d385ff9184feb344e022fe1e7/app/src/main/java/com/itsaky/androidide/tasks/callables/FileTreeCallable.java#L52) for example. It shorts the files alphabetically and then with their type.

You could even use `Comparatior.comparing(Function, Comparator).thenComparing(Function, Comparator)` for even faster sorting. Same goes for below two cases.

Edit:
I see that the below comparators (`Comparator_SIZE` and `Comparator_TYPE`) handle the directory ordering as well. Could you explain why you're trying to order directories here?

Can we not simply use `content.sort(Comparator_SIZE)` here?

**Review by itsaky — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/itsaky/androidide/utils/SortUtils.java (hunk 1 above)
Comment (MrIkso) [reply app/src/main/java/com/itsaky/androidide/utils/SortUtils.java:48]: It's not used class. Add by my mistake

Comment (itsaky) [reply app/src/main/java/com/itsaky/androidide/utils/SortUtils.java:48]: Okay.

**Review by itsaky — APPROVED** [review]

**Comment by itsaky** [comment]
Thank you for your contributions!

**Pull request merged**